wallet/tests/deploy.rs

```diff
@@ -0,0 +1,119 @@
+use serde_json::{json, Value};
+use solana::bank::Bank;
+use solana::cluster_info::Node;
+use solana::db_ledger::create_tmp_ledger_with_mint;
+use solana::fullnode::Fullnode;
+use solana::leader_scheduler::LeaderScheduler;
+use solana::mint::Mint;
+use solana::rpc_request::{RpcClient, RpcRequest, RpcRequestHandler};
+use solana::vote_signer_proxy::VoteSignerProxy;
+use solana_drone::drone::run_local_drone;
+use solana_sdk::bpf_loader;
+use solana_sdk::signature::{Keypair, KeypairUtil};
+use solana_vote_signer::rpc::LocalVoteSigner;
+use solana_wallet::wallet::{process_command, WalletCommand, WalletConfig};
+use std::fs::{remove_dir_all, File};
+use std::io::Read;
+use std::path::PathBuf;
+use std::sync::mpsc::channel;
+use std::sync::{Arc, RwLock};
+use std::thread::sleep;
+use std::time::Duration;
+
+#[test]
+fn test_wallet_deploy_program() {
+    let mut pathbuf = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    pathbuf.push("tests/fixtures/noop");
+    pathbuf.set_extension("so");
+
+    let leader_keypair = Arc::new(Keypair::new());
+    let leader = Node::new_localhost_with_pubkey(leader_keypair.pubkey());
+    let leader_data = leader.info.clone();
+
+    let alice = Mint::new(10_000);
+    let mut bank = Bank::new(&alice);
+    let ledger_path = create_tmp_ledger_with_mint("thin_client", &alice);
+    let entry_height = alice.create_entries().len() as u64;
+
+    let leader_scheduler = Arc::new(RwLock::new(LeaderScheduler::from_bootstrap_leader(
+        leader_data.id,
+    )));
+    bank.leader_scheduler = leader_scheduler;
+    let vote_account_keypair = Arc::new(Keypair::new());
+    let vote_signer =
+        VoteSignerProxy::new(&vote_account_keypair, Box::new(LocalVoteSigner::default()));
+    let last_id = bank.last_id();
+    let server = Fullnode::new_with_bank(
+        leader_keypair,
+        Arc::new(vote_signer),
+        bank,
+        None,
+        entry_height,
+        &last_id,
+        leader,
+        None,
+        &ledger_path,
+        false,
+        None,
+    );
+    sleep(Duration::from_millis(900));
```

Comment:
what happens if we remove this sleep?

Reply:
Works fine locally. I'll push it and see what happens in CI. My recollection is that there was some flakiness in the past that inspired this sleep, but it may not be existent anymore, esp with integration tests run serially.